LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/client/LabClient.java

```diff
@@ -414,22 +402,36 @@ private String getPassword(final String credentialVaultKeyName) throws LabApiExc
 
     @Override
     public boolean resetPassword(@NonNull final String upn) throws LabApiException {
-        // Before calling the api, we need to fetch the secret code from lab vault
-        final String resetApiCode = getSecret(RESET_API_CODE_SECRET_NAME);
-        return resetPassword(upn, resetApiCode);
+        final ResetApi resetApi = new ResetApi();
+        try {
+            final CustomSuccessResponse resetResponse = resetApi.apiResetPut(upn, ResetOperation.PASSWORD.toString());
+            if (resetResponse == null) {
+                return false;
+            }
+
+            final String expectedResult = ("Password reset for user: " + upn).toLowerCase();
+            final boolean result = resetResponse.toString().toLowerCase().contains(expectedResult);
+            if (result) {
+                try {
+                    Thread.sleep(PASSWORD_RESET_WAIT_DURATION);
+                } catch (final InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
```

Comment:
Why are we waiting here?

Reply:
If reset is successful, we give it some time to replicate